### PR TITLE
refactor update/insert/delete qualified name and as name handling

### PR DESCRIPTION
**Pull request by siddontang**
+ Fix issue #369 
+ Unify qualified and as handling for update/insert/delete.  
+ When use as for a table, we can't use the origin table name.

### CODE REVIEW

File: stmt/stmts/insert.go [hunk 1]
```diff
@@ -232,6 +232,12 @@ func (s *InsertIntoStmt) Exec(ctx context.Context) (_ rset.Recordset, err error)
 	}
 
 	insertValueCount := len(s.Lists[0])
+	toUpdateColumns, err0 := checkUpdateColumns(s.OnDuplicate, nil, t)
```
Comment (shenli) [inline stmt/stmts/insert.go]: matchAssignmentToColumn is better than checkUpdateColumns 
I see the result is a map of column index to assignment.

**Comment by shenli** [comment]
LGTM

File: stmt/stmts/update_test.go [hunk 2]
```diff
@@ -212,5 +216,103 @@ func (s *testStmtSuite) TestIssue345(c *C) {
...
 	rows.Close()
+
+	_, err = testDB.Exec(`update t1 as a, t2 set t1.c1 = 10;`)
+	c.Assert(err, NotNil)
+
 	mustCommit(c, tx)
 }
+
+// See https://github.com/pingcap/tidb/issues/369
+func (s *testStmtSuite) TestIssue369(c *C) {
```
Comment (ngaut) [inline stmt/stmts/update_test.go]: Can we simplify the tests ?

Comment (ngaut) [inline stmt/stmts/update_test.go]: Does MyISAM and innodb has the same results in MySQL?

Comment (siddontang) [inline stmt/stmts/update_test.go]: Yes, no different.

Comment (siddontang) [inline stmt/stmts/update_test.go]: Ok, I will simplify it.

**Comment by siddontang** [comment]
PTAL @ngaut

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**